core/src/main/java/hudson/model/Queue.java

```diff
@@ -3009,4 +3011,70 @@ public static Queue getInstance() {
     public static void init(Jenkins h) {
         h.getQueue().load();
     }
+
+    /**
+     * Schedule <tt>Queue.save()</tt> call for near future once items change. Ignore all changes until the time the save
+     * takes place.
+     *
+     * Once queue is restored after a crash, items stages might not be accurate until the next #maintain() - this is not
+     * a problem as the items will be reshuffled first and then scheduled during the next maintainance cycle.
+     *
+     * Implementation note: Queue.load() calls QueueListener hooks for every item deserialized that can hammer the persistance
+     * on load. The problem is avoided by delaying the actual save for the time long enough for queue to load so the save
+     * operations will collapse into one. Also, items are persisted as buildable or blocked in vast majority of cases and
+     * those stages does not trigger the save here.
+     */
+    @Extension
+    @Restricted(NoExternalUse.class)
+    public static final class Saver extends QueueListener implements Runnable {
+        private static final int DISABLED = -1;
+        private static final int MIN_DELAY = 10;
+
+        @VisibleForTesting
+        /*package*/ static /*final*/ int DELAY_SECONDS = 60;
```

Comment:
Cleaner to have an initializer method for this I think (easy to unit-test that logic trivially), and would prefer to have a public getter so other APIs can see how often the Queue writes -- not required.